blockchain/chain_sync/src/sync.rs

```diff
@@ -193,6 +195,40 @@ where
                             )
                             .await
                     }
+                    Some(NetworkEvent::PubsubMessage { source, message }) => {
+                        match message {
+                            forest_libp2p::PubsubMessage::Block(b) => {
+                                info!("Received block over GossipSub: {} from {}", b.header.epoch(), source.clone().unwrap());
```

Comment:
You're calling the source clone and unwrap three times in this block, consider doing once or handling this better (I get that the unwrap should not fail, but doesn't mean it shouldn't be handled)